### ISSUE TITLE
$user->hasRole returns null in middleware, while tinker is returning correct value

### ISSUE THREAD

**Issue by amarkulo** [spatie/laravel-permission#1296]
I have created very simple Laravel project with Backpack, role and permissions seeders and nothing else. 

When I log in, dashboard is shown like it should, but If I try to change CheckIfAdmin middleware to return if the user is admin or not with $user->hasRole('admin') I always get false. 

Tinker is returning true for the same (and only) user.

```
>>> $a = User::first()
=> App\User {#3204
     id: 3,
     name: "Admin",
     email: "admin@example.com",
     email_verified_at: null,
     created_at: "2019-12-09 19:43:27",
     updated_at: "2019-12-09 19:43:27",
   }
>>> $a->hasRole('admin')
=> true
>>> $a->roles
=> Illuminate\Database\Eloquent\Collection {#3228
     all: [
       Spatie\Permission\Models\Role {#3202
         id: 1,
         name: "admin",
         guard_name: "web",
         created_at: "2019-12-09 19:40:37",
         updated_at: "2019-12-09 19:40:37",
         pivot: Illuminate\Database\Eloquent\Relations\MorphPivot {#3209
           model_id: 3,
           role_id: 1,
           model_type: "App\User",
         },
       },
     ],
   }
>>>
```

For some reason, roles for user are always returned as null, that's why this is failing.

What am I missing?

**Comment by drbyte** [comment]
1. This sounds like a Backpack question, since you're dealing with Backpack's implementation of this package.
2. I'm guessing that Backpack is using guards other than `web` which is what Tinker is running into because you've not specified a guard to check against, and `web` is likely listed as the first and default guard in your auth.php config file.

Having not seen the CheckIfAdmin middleware code you mentioned, nor your alterations, it's hard to be certain, but I'm guessing you're just not being consistent with which guard you're testing against.

**Comment by amarkulo** [comment]
I have solved the problem. 

CheckIfAdmin middleware was just returning this which in itself was not a problem:

```
    private function checkIfUserIsAdmin($user)
    {
        return $user->hasRole('admin');
    }
```

But the problem was like you said, wrong guard name. In auth.php default guard was "web", while Backpack is using it own guard "backpack", so when I created roles with seeder, it created them with default web guard. Solution to create them properly and keep both guards was to specify which guard should both roles and permissions be created with.

```
Role::create(['name' => 'admin', 'guard_name' => backpack_guard_name()]);
Permission::create(['name' => 'Can view dashboard, 'guard_name' => backpack_guard_name()]);
```

Thanks a lot for help and pointing me in the right way!
